utils/utils.go

```diff
@@ -1523,12 +1523,14 @@ func CreateConfigMap(mapName string) {
 
 	operatorNs, _ := GetOperatorNamespace()
 	if operatorNs == "" {
-		// This should only happen when running locally in development
-		// Probably best to just return. The operator global config map is tried
-		// again in the reconcile loop, and don't want to duplicate logic to
-		// guess what the namespace should be
-		utilsLog.Info("Couldn't create operator config map as operator namespace was not found")
-		return
+		// This could happen if the operator is running locally, i.e. outside the cluster
+		watchNamespaces, err := GetWatchNamespaces()
+		if err != nil {
+			utilsLog.Error(err, "Error getting watch namespace")
+			return
+		}
+		// If the operator is running locally, use the first namespace in the `watchNamespaces`
+		operatorNs = watchNamespaces[0]
 	}
 	configMap := &corev1.ConfigMap{}
 	err := client.Get(context.TODO(), types.NamespacedName{Name: mapName, Namespace: operatorNs}, configMap)
```

Comment:
It'll be good to check that the error was `IsNotFound` ([example](https://github.com/application-stacks/runtime-component-operator/blob/4ac56bcc9077dd495d87931cfb5bbd26539f2d50/utils/reconciler.go#L135)) and only proceed to create ConfigMap if so. Otherwise, we could update the ConfigMap with default settings and override any user-specified configuration when we encounter any error

So changing `if err != nil {` --> `if err != nil && apierrors.IsNotFound(err) {`